kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -50,17 +53,40 @@ define([
             bus = runtime.bus().connect(),
             channel = bus.channel(config.channelName),
             ui,
-            dataSource = 'ftp_staging', // only option now, should come from the parameterSpec.
+            dd_options = spec.original.dynamic_dropdown_options,
+            dataSource = dd_options.data_source || 'ftp_staging',
             model = {
                 value: undefined
             },
             stagingService = new StagingServiceClient({
                 root: runtime.config('services.staging_api_url.url'),
                 token: runtime.authToken()
             }),
+            searchClient = new KBaseSearchEngine(
+                runtime.config('services.KBaseSearchEngine.url'),
+                {token: runtime.authToken()}
+            ),
             userId = runtime.userId(),
             eventListeners = [];
 
+        var flattenObject = function(ob) {
```

Comment:
What's the function do? Does it just take a nested object and flattens it into key-value pairs? I don't quite follow how it's being used.

Reply:
Added a docstring to explain it's function